kyuubi-zookeeper/src/main/scala/org/apache/kyuubi/zookeeper/ZookeeperConf.scala

```diff
@@ -49,6 +49,15 @@ object ZookeeperConf {
       .stringConf
       .createOptional
 
+  val ZK_CLIENT_USE_HOSTNAME: ConfigEntry[Boolean] =
+    buildConf("kyuubi.zookeeper.embedded.client.use.hostname")
+      .doc("When true, embedded Zookeeper prefer to bind hostname, otherwise, ip address. " +
+        "Note that, the default value is set to `false` when engine running on Kubernetes " +
+        "to prevent potential network issues.")
+      .version("1.7.2")
+      .booleanConf
+      .createWithDefault(true)
```

Comment:
use true to keep behavior, but maybe false is better? since it's not intend to use for production, behavior change is fine.